docs/StrictProvenance.md

```diff
@@ -0,0 +1,126 @@
+# StrictProvenance Architectures
```

Comment:
Can you add a summary at the start of the file that says everything in paragraph.  Perhaps, something like:
  
>  To support novel architectures, such as CHERI, `snmalloc` uses an `AuthPtr` wrapper class that tracks the way in which this pointer can be used.
>  You can view the annotation on `AuthPtr` as specifying the type of operations that are supported by this pointer such as 
> *  address arithmetic within a certain range, i.e. slab.
> *  if virtual memory mappings maybe modified
>
> This restrictions are not imposed on most platforms, but on CHERI they can be enforced by hardware.  The rest of the document...

Reply:
I would also perhaps put the more background material later in the document, and focus on what someone modifying snmalloc needs to know at the start, which is kind of the opposite way round to you have it.

Who is this document for, and why are they reading it?  My imagination that most people reading md files on github: are trying to work out how code works, or how to change it.